packages/lodestar/src/network/reqresp/response/rateLimiter.ts

```diff
@@ -0,0 +1,99 @@
+import {ILogger} from "@chainsafe/lodestar-utils";
+import PeerId from "peer-id";
+import {MapDef} from "../../../util/map";
+import {IPeerRpcScoreStore, PeerAction} from "../../peers/score";
+import {IRateLimiter} from "../interface";
+import {RateTracker} from "../rateTracker";
+
+interface IRateLimiterModules {
+  logger: ILogger;
+  peerRpcScores: IPeerRpcScoreStore;
+}
+
+export type RateLimiterOpts = {
+  requestCountTotalLimit: number;
+  requestCountPeerLimit: number;
+  blockCountTotalLimit: number;
+  blockCountPeerLimit: number;
+  rateTrackerTimeoutMs: number;
+};
+
+/**
+ * The rate tracker for all peers.
+ */
+export class InboundRateLimiter implements IRateLimiter {
+  private readonly logger: ILogger;
+  private readonly peerRpcScores: IPeerRpcScoreStore;
+  private requestCountTotalTracker: RateTracker;
```

Comment:
I don't believe this one is necessary since serving non-blocks request _should_ be cheap